test/pipelines/unit/test_pipeline.py

```diff
@@ -317,3 +317,20 @@ def test_from_dict_without_connection_receiver():
         Pipeline.from_dict(data)
 
     err.match("Missing receiver in connection: {'sender': 'some.sender'}")
+
+def test_falsy_connection():
+    @component
+    class A:
+        @component.output_types(y=int)
+        def run(self, x: int):
+            return {"y": 0}
+    @component
+    class B:
+        @component.output_types(y=int)
+        def run(self, x: int):
+            return {"y": x}
```

Comment:
You can use the `component_class` factory to do this.
https://github.com/deepset-ai/canals/blob/main/canals/testing/factory.py#L9

Reply:
Thanks, I have changed that.